agreementbot/secrets/vault/init.go

```diff
@@ -48,7 +48,9 @@ func (vs *AgbotVaultSecrets) Login() (err error) {
 		return errors.New(fmt.Sprintf("agbot unable to login, error: %v", err))
 	}
 
-	defer resp.Body.Close()
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
```

Comment:
Please move this to after line 47, right after invokeVaultWithRetry because the response body could be non-nil even in the error case.

Reply:
The same comment for the rest of the changes.

Reply:
Updated PR, thanks!